pkg/mcp/kubernetes.go

```diff
@@ -506,26 +543,120 @@ func (k *kubernetesBackend) restartServer(ctx context.Context, id string) error
 		return fmt.Errorf("failed to get deployment %s: %w", id, err)
 	}
 
+	// Fetch K8s settings
+	k8sSettings, err := k.k8sSettings.GetK8sSettings(ctx)
+	if err != nil {
+		// Log error but continue with defaults
+		log.Warnf("Failed to get K8s settings, using defaults: %v", err)
+		k8sSettings = nil
+	}
+
+	// Compute K8s settings hash
+	k8sSettingsHash := computeK8sSettingsHash(k8sSettings)
+
+	// Build the patch with restart annotation and k8s settings hash
+	podAnnotations := map[string]string{
+		"kubectl.kubernetes.io/restartedAt": time.Now().Format(time.RFC3339),
+		"obot.ai/k8s-settings-hash":         k8sSettingsHash,
+	}
+
+	// Update the deployment metadata annotation as well
+	deploymentAnnotations := map[string]string{
+		"obot.ai/k8s-settings-hash": k8sSettingsHash,
+	}
+
+	// Build the patch structure
 	patch := map[string]any{
+		"metadata": map[string]any{
+			"annotations": deploymentAnnotations,
+		},
 		"spec": map[string]any{
 			"template": map[string]any{
 				"metadata": map[string]any{
-					"annotations": map[string]string{
-						"kubectl.kubernetes.io/restartedAt": time.Now().Format(time.RFC3339),
-					},
+					"annotations": podAnnotations,
 				},
+				"spec": map[string]any{},
 			},
 		},
 	}
 
+	templateSpec := patch["spec"].(map[string]any)["template"].(map[string]any)["spec"].(map[string]any)
```

Comment:
nit: if you set `templateSpec := make(map[string]any)` above the patch, then you can set `"spec": templateSpec` in the patch and not do all the casting.